draft-selander-lake-authz.md

```diff
@@ -535,30 +535,45 @@ V receives the voucher response from W over the secure connection. If the receiv
 # REST Interface at W
 
 The interaction between V and W is enabled through a RESTful interface exposed by W.
+This RESTful interface MAY be implemented using either HTTP or CoAP.
 V SHOULD access the resources exposed by W through the protocol indicated by the scheme in LOC_W URI.
-In case the scheme indicates "https", V SHOULD perform a TLS handshake with W and use HTTP.
+
+## Scheme "https" {#scheme-https}
+In case the scheme indicates "https", V MUST perform a TLS handshake with W and use HTTP.
+If the public key of V, PK_V, is a signature key that can be used in a TLS handshake, V SHOULD use the certificate containing PK_V to authenticate to W as a client.
+In case the public key of V, PK_V, is a static DH key that cannot be used in a TLS handshake, V MUST first perform a TLS handshake with W using the available compatible keys.
+V MUST then perform an EDHOC handshake on top of the TLS session, using and proving the possesion of PK_V.
+Performing the EDHOC handshake is only necessary if V did not use PK_V in the TLS handshake with W.
+
+## Scheme "coaps"
 In case the scheme indicates "coaps", V SHOULD perform a DTLS handshake with W and access the same resources using CoAP.
-In both cases, V MUST perform client authentication to authenticate to W, using a certificate containing the PK_V public key.
+The normative requirements on performing the DTLS and EDHOC handshakes from {{scheme-https}} remain the same, except that TLS is replaced with DTLS.
 
-## HTTP URIs
+## Scheme "coap"
+In case the scheme indicates "coap", V SHOULD perform an EDHOC handshake with W, as specified in {{Appendix A of I-D.ietf-lake-edhoc}} and access the same resources using OSCORE and CoAP.
+The authentication credential in this EDHOC run MUST be the certificate containing PK_V.
```

Comment:
NEW
The authentication credential in this EDHOC run MUST be CRED_V.

Reply:
fixed